src/nvim/highlight.c

```diff
@@ -927,10 +927,11 @@ int object_to_color(Object val, char *key, bool rgb, Error *err)
   } else if (val.type == kObjectTypeString) {
     String str = val.data.string;
     // TODO(bfredl): be more fancy with "bg", "fg" etc
+    if (!str.size || STRICMP(str.data, "NONE") == 0) {
```

Comment:
This also clears the hl group for an empty string (`""`), so let's capture that in a test case too